server/src/main/java/org/opensearch/search/suggest/completion/CompletionSuggester.java

```diff
@@ -124,6 +124,7 @@ private static void suggest(IndexSearcher searcher, CompletionQuery query, TopSu
                 }
             }
         }
+        collector.finish();
```

Comment:
@nknize sorry,  to reconfirm, you find it is it required? The leaf collector completion should be enough (I think) https://github.com/opensearch-project/OpenSearch/pull/8967/files#diff-07393079d934d51c3ca48de83a83998bbf843f54b05c29f0235a9a7115b2286dR123, thank you.

Reply:
It shouldn't be, but I hit a random test failure. Possibly before you committed #8968. Removing and retrying.